Expo1900/Expo1900/Controller/View/ViewController.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  Expo1900 - ViewController.swift
+//  Created by yagom. 
+//  Copyright © yagom academy. All rights reserved.
+// 
+
+import UIKit
+
+class ViewController: UIViewController {
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var visitorsLabel: UILabel!
+    @IBOutlet weak var locationLabel: UILabel!
+    @IBOutlet weak var durationLabel: UILabel!
+    @IBOutlet weak var descLabel: UILabel!
+    @IBOutlet weak var itemsbutton: UIButton!
+    
+    var expositionUniverselles: ExpositionUniverselle = ExpositionUniverselle(title: "", visitors: 0, location: "", duration: "", desc: "")
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        jsonPaser()
+        
+        titleLabel.text = expositionUniverselles.title
+        visitorsLabel.text = String(expositionUniverselles.visitors)
+        locationLabel.text = expositionUniverselles.location
+        durationLabel.text = expositionUniverselles.duration
+        descLabel.text = expositionUniverselles.desc
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.navigationController?.isNavigationBarHidden = true
+    }
+    
+    func jsonPaser() {
+        let jsonDecoder = JSONDecoder()
+        guard let expositionUniversellesData: NSDataAsset = NSDataAsset(name: "exposition_universelle_1900") else {
+            return
+        }
+        do {
+            self.expositionUniverselles = try jsonDecoder.decode(ExpositionUniverselle.self, from: expositionUniversellesData.data)
+        } catch {
+            print(error.localizedDescription)
```

Comment:
고민해보시면 좋을 것 같아서 질문!

여기서 실패처리가 하나는 옵셔널이고 하나는 에러인데 어떤 차이가 있을까요?

Reply:
위의 NSDataAsset의 경우 에셋 안에 있는 데이터 중 이름이 동일한 파일을 가져와 사용하기 때문에 
" 이름이 일치하지 않은 = 파일이 없는 = nil의 경우 " 를 위해 옵셔널 처리를 해주었습니다.

에러처리의 경우 jsonDecoder의 정의를 보니까 init 과정에서 throw를 낼 수 있어서 do catch 문을 사용했습니다.

Reply:
아하 해당 문법을 사용하신 이유가 아니라 음
왜 어떤 메소드들은 실패처리에 대해 에러를 던지게 되어있고 어떤 메소드는 옵셔널로 실패를 처리하는 걸까? 라는 의문을 가져보셨으면 좋을 것 같아서 던져본 질문이었어요

Reply:
```suggestion
        guard let expositionUniversellesData: NSDataAsset = NSDataAsset(name: "exposition_universelle_1900") else {
            return
        }
        do {
            self.expositionUniverselles = try jsonDecoder.decode(ExpositionUniverselle.self, from: expositionUniversellesData.data)
        } catch {
            print(error.localizedDescription)
```

아하! 문법적인 것 이전에 다시 생각해보니까 옵셔널을 통해 데이터를 가져왔을 때 옵셔널이 성공적으로 해제 됐다면 빈 데이터가 아니므로 decode 과정에서 남은건 성공 여부만 남았다고 생각해요 ( nil 값 검증은 guard let을 통해 해결 )

Reply:
음 제가 의도를 잘 전달하지 못한 것 같아요
음 `decode(_, from:)` 메소드는 왜 실패로 Error를 던지고
`NSDataAsset(name: )` 은 생성실패에서 에러를 안던지고 옵셔널타입일까 어떤차이일까? 라는 질문이었어요